package.json

```diff
@@ -139,14 +133,15 @@
     "eslint-plugin-spellcheck": "^0.0.17",
     "execa": "^5.1.1",
     "glob": "^7.2.0",
-    "husky": "^3.1.0",
+    "husky": ">=6",
     "istanbul-lib-coverage": "^3.2.0",
     "istanbul-lib-report": "^3.0.0",
     "istanbul-reports": "^3.1.5",
     "jest": "^29.0.0",
     "jest-circus": "^29.0.0",
     "jest-junit": "^12.0.0",
     "js-yaml": "^4.0.0",
+    "lint-staged": ">=10",
```

Comment:
Why is this package needed? Same comment about the `>=` versioning as above.

Reply:
here's the doc https://prettier.io/docs/en/precommit.html#option-1-lint-stagedhttpsgithubcomokonetlint-staged

Reply:
Can prettier be executed in the existing [pre-commit](https://github.com/aws-amplify/amplify-cli/blob/dev/package.json#L88) script? I tend to be conservative about adding dependencies and devDependencies.

Reply:
Yes, prettier can be done at pre-commit auto-magically. This way we don't have to wait until the PR checker fails. 

Also, the change in this PR will allow VS Code to format on save.

Reply:
We can't use that because 
1. `yarn prettier --write && yarn eslint --cache --fix` doesn't work directly in pre-commit
2. even though it would work, we only want to lint the staged files, which is much quicker

If not using `lint-staged`, we could write some script mentioned [here](https://prettier.io/docs/en/precommit.html#option-6-shell-script), but I think it could have unknown security issue as well. I would rather have Depend-bot warns me the issue.

Reply:
We have those scripts already defined in the root package.json. I should have said `yarn prettier-changes && yarn lint-fix`

Reply:
Right, but they run through the entire repo and they don't work at the pre-commit script. 

Maybe I missed something, did you test it works on your end?

Reply:
Those commands work on my end and don't run through the entire repo, but if they're not working for you then we perhaps the new package is the better approach.

Reply:
That's a good point. I just updated the PR and removed `lint-staged` 🙌

Reply:
Note:  the pre-commit script didn't work on my end was because `lint-stage` changed my git hook path to `..husky`. 

```sh
$ git config core.hooksPath
> .husky
```